upgrade/upgrade-scripts/upgrade-container

```diff
@@ -149,6 +151,8 @@ function create_upgrade_container() {
 			"$ROOTFS_DATASET@$SNAPSHOT_NAME" "rpool/ROOT/$CONTAINER"
 		copy_optional_dataset_property "$PROP_HOTFIX_VERSION" \
 			"$ROOTFS_DATASET@$SNAPSHOT_NAME" "rpool/ROOT/$CONTAINER"
+		copy_optional_dataset_property "$PROP_MINIMUM_VERSION" \
```

Comment:
Why are we using `copy_optional` here, rather than `copy_required`?

Presumably this is to handle the case of moving from a version that didn't have the field, to a version that does have the field; is that correct?

If this is the only consideration, then I'd rather not use the `optional` function for all delphix versions, even for versions that we expect the field to be there.. I'd suggest we tweak this such that we only attempt the copy when coming from versions where we expect the field to exist, and then simply use the `required` function.. this way, if we're on a version that should have the field, but it doesn't, we'll properly report the error..

I think we can easily do this.. e.g. something like this:
```
if compare_versions "$(get_current_version)" ge "6.0.11.0"; then
    copy_required_dataset_property ...
fi
```

What do you think?